presto-hive/src/main/java/com/facebook/presto/hive/HiveManifestUtils.java

```diff
@@ -85,4 +99,53 @@ public static long getFileSize(Page statisticsPage, int position)
         }
         return Optional.of(manifestBuilder.build());
     }
+
+    public static Map<String, String> updatePartitionMetadataWithFileNamesAndSizes(PartitionUpdate partitionUpdate, Map<String, String> metadata)
+    {
+        ImmutableMap.Builder<String, String> partitionMetadata = ImmutableMap.builder();
+        List<FileWriteInfo> fileWriteInfos = new ArrayList<>(partitionUpdate.getFileWriteInfos());
+
+        // Sort the file infos based on fileName
+        fileWriteInfos.sort(Comparator.comparing(info -> Integer.valueOf(info.getWriteFileName())));
+
+        // Join the file names into a consolidated string
+        String fileNames = joinFileNames(fileWriteInfos.stream().map(FileWriteInfo::getWriteFileName).collect(toImmutableList()));
+
+        // Join the file sizes
+        String fileSizes = Joiner.on(COMMA).join(fileWriteInfos.stream()
+                .map(FileWriteInfo::getFileSize)
+                .filter(Optional::isPresent)
+                .map(Optional::get)
```

Comment:
- When will we have optional sizes? Will this cause inconsistency between file names and sizes?
- Suggest using delta encoding. Or more easily, build the string and zstd it.